src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -99,22 +99,26 @@ internal static bool IsOptionName(string shortOptionName, string longOptionName,
         internal static bool IsOptionName(string optionName, ReadOnlySpan<char> value)
         {
             Debug.Assert(isAllAscii(optionName.AsSpan()));
-            if (isAllAscii(value))
+
+            if (optionName.Length != value.Length)
+                return false;
+
+            for (int i = 0; i < optionName.Length; i++)
             {
-                if (optionName.Length != value.Length)
-                    return false;
+                char ch = value[i];
+                if (ch > 127)
+                {
+                    // If a non-ascii character is encountered, do an InvariantCultureIgnoreCase comparison
+                    return optionName.AsSpan().Equals(value, StringComparison.InvariantCultureIgnoreCase);
+                }
 
-                for (int i = 0; i < optionName.Length; i++)
+                if (optionName[i] != char.ToLowerInvariant(ch))
                 {
-                    if (optionName[i] != char.ToLowerInvariant(value[i]))
-                    {
-                        return false;
-                    }
+                    return false;
                 }
-                return true;
             }
 
-            return optionName.AsSpan().Equals(value, StringComparison.InvariantCultureIgnoreCase);
+            return true;
 
             static bool isAllAscii(ReadOnlySpan<char> span)
```

Comment:
Looks like this is now only used in an assert, consider wrapping it in `#if DEBUG`